src/carp.cc

```diff
@@ -73,24 +68,16 @@ carpInit(void)
         if (p->weight == 0)
             continue;
 
-        ++n_carp_peers;
+        rawCarpPeers.push_back(p);
 
         W += p->weight;
     }
 
-    if (n_carp_peers == 0)
+    if (rawCarpPeers.empty())
         return;
 
-    carp_peers = (CachePeer **)xcalloc(n_carp_peers, sizeof(*carp_peers));
-
     /* Build a list of the found peers and calculate hashes and load factors */
-    for (P = carp_peers, p = Config.peers; p; p = p->next) {
-        if (!p->options.carp)
-            continue;
-
-        if (p->weight == 0)
-            continue;
-
+    for (const auto &p: rawCarpPeers) {
```

Comment:
Since these are raw pointers, it is "cheaper" (and simpler) to copy them than to reference them.

```suggestion
    for (const auto p: rawCarpPeers) {
```